pkg/publish/ingress.go

```diff
@@ -21,7 +23,15 @@ import (
 
 func toPrefix(domain, serviceName string, appInstance *v1.AppInstance) (hostPrefix string) {
 	if strings.HasSuffix(domain, "on-acorn.io") {
-		appInstanceIDSegment := strings.SplitN(string(appInstance.GetUID()), "-", 2)[0]
+		appInstanceSplitName := strings.Split(appInstance.GetName()+"-"+serviceName, "-")
```

Comment:
You did this join + split based on the example Darren and I gave you:
```
default-wispy-smoke-sha256(default:wispy:smoke)[:12] 
```
but now that I see it, I realize it's wrong. We dont want to split the appInstance name. We want to keep it whole and just use the `:` separator between the distinct user-controllable parts, so that any `-` _within_ those parts are maintained for the hashing.

So, the example should actually read:
```
default-wispy-smoke-sha256(default:wispy-smoke)[:12] 
```

Can you fix?

Reply:
should be fixed now

Reply:
The code is still splitting a piece it shouldn't you don't need to split "serviceName" at all. That is the name of the "container" in the acornfile and we want to keep it whole.

Reply:
basically, you just need to has `serviceName + ":" + appInstance.GetName()`